test/fakeintake/server/server_test.go

```diff
@@ -41,6 +41,12 @@ func testServer(t *testing.T, opts ...func(*Server)) {
 		assert.Empty(t, fi.URL())
 	})
 
+	t.Run("should use the hard coded port", func(t *testing.T) {
+		opts = append(opts, WithAddress("localhost:1234"))
+		fi := NewServer(opts...)
+		assert.Equal(t, "localhost:1234", fi.server.Addr)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
We should test it avoiding accessing internals. We should do a `Get` request on the new port and on the old port.